Adding graphql_client as dependency in Router Plugin causes internal dependency issue

**Describe the bug**
I am writing a plugin to use the graphql client to make a query call for authorization but when I add graphql_client as a dependency, it breaks the the code in https://github.com/apollographql/rover/blob/geal/launchpad/crates/launchpad/src/introspect/runner.rs

**To Reproduce**
Steps to reproduce the behavior:
1. set up context example project -> https://github.com/apollographql/router/tree/v0.1.0-preview.6/examples/context
2. add these as dependency in Cargo.toml 
````
graphql_client = { version = "0.10.0", features = ["reqwest-blocking"]}
reqwest = { version = "0.11.10", features = ["blocking"]}
````
3. run `cargo build`
4. See error

**Expected behavior**
cargo build compiles without any issues

**Output**
````
  Compiling launchpad v0.1.0 (https://github.com/apollographql/rover.git?rev=94141242ba34cf00cde9630fc4a6dcd05d4fa5da#94141242)
error[E0433]: failed to resolve: use of undeclared type `HeaderMap`
  --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:30:26
   |
30 |     let mut header_map = HeaderMap::new();
   |                          ^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  | use reqwest::header::HeaderMap;
   |

error[E0433]: failed to resolve: use of undeclared type `HeaderName`
  --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:33:13
   |
33 |             HeaderName::from_bytes(header_key.as_bytes())?,
   |             ^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  | use reqwest::header::HeaderName;
   |

error[E0433]: failed to resolve: use of undeclared type `HeaderValue`
  --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:34:13
   |
34 |             HeaderValue::from_str(&header_value)?,
   |             ^^^^^^^^^^^ not found in this scope
   |
help: consider importing this struct
   |
1  | use reqwest::header::HeaderValue;
   |

error[E0659]: `reqwest` is ambiguous
 --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:6:5
  |
6 | use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
  |     ^^^^^^^ ambiguous name
  |
  = note: ambiguous because of multiple potential import sources
  = note: `reqwest` could refer to a crate passed with `--extern`
  = help: use `::reqwest` to refer to this crate unambiguously
note: `reqwest` could also refer to the module imported here
 --> /Users/macuser/.cargo/git/checkouts/rover-efd9f422be37a06b/9414124/crates/launchpad/src/introspect/runner.rs:5:5
  |
5 | use graphql_client::*;
  |     ^^^^^^^^^^^^^^^^^
  = help: use `self::reqwest` to refer to this module unambiguously

Some errors have detailed explanations: E0433, E0659.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `launchpad` due to 4 previous errors
````

**Additional context**
Dependency tree is apollo router -> uplink -> rover (branch = geal/launchpad)

--- a/uplink/Cargo.toml
+++ b/uplink/Cargo.toml
@@ -8,7 +8,7 @@ build = "build.rs"
 
 [dependencies]
 futures = "0.3.21"
-graphql_client = { version="0.10.0" }
+graphql_client = { version = "0.10.0" }
 reqwest = { version = "0.11.10", default_features = false, features = [
     "rustls-tls",
     "json",
@@ -20,7 +20,7 @@ tracing = "0.1.34"
 url = "2.2.2"
 
 [build-dependencies]
-launchpad = { git = "https://github.com/apollographql/rover.git", rev = "94141242ba34cf00cde9630fc4a6dcd05d4fa5da" }
+launchpad = { git = "https://github.com/apollographql/rover.git", rev = "c52bd32f1493e63844fd74e1a1d32f18e5c88394" }
 reqwest = { version = "0.11.10", default_features = false, features = [
     "rustls-tls",
 ] }

